src/libgit2/repository.c

```diff
@@ -582,10 +582,14 @@ static int validate_ownership_cb(const git_config_entry *entry, void *payload)
 		         strncmp(test_path, "//wsl.localhost/", strlen("//wsl.localhost/")) != 0)
 			test_path++;
 #endif
-
-		if (git_fs_path_prettify_dir(&data->tmp, test_path, NULL) == 0 &&
```

Comment:
I think that instead of `prettify`, we need only call `path_to_dir`.

Reply:
I don't think that using `path_to_dir` is the right way. This would accept `safe.directory` entries ending with a slash which vanilla Git does not accept.

Reply:
Is that actually a problem, though? We also accept `//wsl.localhost/...` prefixed paths which Git for Windows doesn't accept.

Reply:
From my point of view, I would keep it as consistent as possible.

Reply:
Let's do one fix at the time. It's ok for me if we take your suggested patch.

Reply:
> From my point of view, I would keep it as consistent as possible.

OK. Let's do that. This feels like a (minor) bug in git that it doesn't allow trailing slashes, but I've often said that libgit2's goal is to be "bug for bug compatible".

Reply:
I updated the branch and hope that I did not forget anything.

Reply:
@ethomson 
I updated the branch and hope that I did not forget anything.